### PR TITLE
Setup Authorization

### PR DESCRIPTION
**Pull request by patriciachrysy**
Install CanCanCan in your project.
Add a role column to the users table. Remember to use a migration for this.
A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -7,7 +7,13 @@
...
     </div>
     <p><%= @post.text %></p>
-    <%= link_to "like", user_post_likes_path(@post.author, @post), class: 'like-button' %>
+    <div class="post-action-buttons">
+        <%= link_to "like", user_post_likes_path(@post.author, @post), class: 'like-button' %>
+        <%= link_to "Delete", user_post_path(@post.author, @post), data: {
+                turbo_method: :delete,
+                turbo_confirm: 'Are you sure you want to delete this post?'
+            }, class: 'like-button' if can? :destroy, Post %>
+    </div>
```
Comment (julie-ify) [inline app/views/posts/show.html.erb]: - [ ] The delete button for post is displayed regardless of the user's role. Kindly note that only the creator of the post and the admin should be able to see the delete buttons at all. Kindly fix your views.

- [ ] Your post delete functionality does not work at all. When I click it as the admin or post owner, nothing happens - the post is not deleted. Kindly fix that.

File: app/views/posts/_comment_thumb.html.erb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 <div class="comment-content">
     <span><%= commenter %>:</span>
     <span><%= comment %></span>
+    <%= link_to "Delete", user_post_comment_path(@post.author, @post, comment_id), data: {
+            turbo_method: :delete,
+            turbo_confirm: 'Are you sure you want to delete this comment?'
+        }, class: 'like-button' if can? :destroy, Comment %>
```
Comment (julie-ify) [inline app/views/posts/_comment_thumb.html.erb]: - [ ] The delete button for comments is displayed regardless of the user's role. Kindly note that only the creator of the comment and the admin should be able to see the delete buttons at all. Kindly fix your views.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @patriciachrysy and @jkanyi-web,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 3]
```diff
@@ -20,6 +20,10 @@ def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
 
+  def is?(requested_role)
+    role == requested_role.to_s
+  end
+
```
Comment (Reem-lab) [inline app/models/user.rb:26]: - Kudos for creating create their user.admin? method inside the user model. 👍🏻🎉

**Review by Reem-lab — APPROVED** [review]
# Hi @patriciachrysy and @jkanyi-web,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  No Linters error ✅
- [ ]  Kudos for working on the authorization rules🤩
- [ ] Good job working on the requested changes 💪🏻
- [ ] Your design is amazing ⭐


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**